src/main/java/org/hydev/mcpm/client/uninstallUseCase/Uninstaller.java

```diff
@@ -0,0 +1,36 @@
+package org.hydev.mcpm.client.uninstallUseCase;
+import java.io.File;
+import java.util.Arrays;
+import org.hydev.mcpm.client.injector.PluginLoader;
+import org.hydev.mcpm.client.injector.PluginNotFoundException;
+
+
+public class Uninstaller implements UninstallBoundary{
+    private boolean remove_depen;
+
+
+    public void uninstall(String name) throws PluginNotFoundException {
+        PluginLoader pil = new PluginLoader();
+        pil.unloadPlugin(name);
+
+        // Find plugin file
+        // could be useful for removing dependencies (?)
+        var plugins_folder = new File("plugins");
+        var plugins = Arrays.stream((plugins_folder.listFiles()))
+                .filter(plugin -> plugin.getName().endsWith(name+".jar"));
+
+        File file = new File(plugins.toArray()[0].toString());
+
+        if (file.delete()) {
+            System.out.println("File deleted successfully");
+        } else {
+            System.out.println("Failed to delete the file");
+        }
+
+
+//        if (remove_depen){
```

Comment:
Just for reference, `PluginYML` has a `depend` array containing the names of all the plugins it depends on. Before removing them outright(!) though I'd check PluginTracker to see if any of the dependencies are in `listManuallyInstalled` (because if the user installed them we definitely don't want to get rid of them as like a tangent).

Reply:
You also want to make it recursive, in case the dependencies have dependencies
Your base case would be if the plugin is not installed, then you don't have to do recursion.
Although this wouldn't work if you manually uninstalled something and wanted to actually uninstall all its plugins.
In that case you probably want a visited array in case of circular dependencies, then the base case would be to return if you've already visited the plugin.

Reply:
Iterative works as well, the benefit to iterative is that you may not have to make a helper recursive function, but also could be harder to implement. lmk if I can help